### ISSUE TITLE
[Issue]: No matching subtitles found

### ISSUE THREAD

**Issue by williameyelash** [MichaelYochpaz/iSubRip#43]
### Confirmations

- [X] I have checked the issues tab, and couldn't find an existing open issue for the issue I want to report.

### OS Type

Windows

### Python Version

3.11.1

### Package Version

2.4.0

### Description

https://tv.apple.com/fi/movie/goldstone/umc.cmc.5up0cp30rbmrod9hc1xjq86o3?playableId=tvs.sbd.9001%3A1501840635

"No matching subtitles found". Worked just fine with the previous version.

**Comment by williameyelash** [comment]
Hm, no, I just upgraded via command line and that's it. Uninstalled it now and installed again -- same problem.

**Comment by williameyelash** [comment]
Went back to 2.3.3 and it works fine for me.

**Comment by MichaelYochpaz** [comment]
I'm going to need some additional information as I'm unable to reproduce this issue.

1. Add the complete output when using v2.4.1 on the same URL (like the one I sent [here](https://github.com/MichaelYochpaz/iSubRip/issues/43#issuecomment-1564078670)).
2. Does this issue reproduce when downloading other content after the update? Or just this specific movie?
3. Run `isubrip https://tv.apple.com/ca/movie/living/umc.cmc.7jo8l4dfds2vupar7bidbklbe` and add the complete output (I want to see if it filters any specific languages, and if it does, which ones).
4. Create the following config on `%USERPROFILE%\.isubrip\config.toml` (since you don't have one, you'll have to create a new file):
    ```toml
    [downloads]
    languages = []
    ```
    Run the command again (on v2.4.1), and add the complete output of that as well.

**Comment by williameyelash** [comment]
1. Scraping https://tv.apple.com/fi/movie/goldstone/umc.cmc.5up0cp30rbmrod9hc1xjq86o3?playableId=tvs.sbd.9001%3A1501840635
Found movie: Goldstone (2017)
No matching subtitles were found.

It then creates a .zip file with the size of 1 KB that you can't unzip, it says "No files to extract"
2. No, it's relevant for all links.
3. Scraping https://tv.apple.com/ca/movie/living/umc.cmc.7jo8l4dfds2vupar7bidbklbe
Found movie: Living (2023)
No matching subtitles were found.
4. Scraping https://tv.apple.com/ca/movie/living/umc.cmc.7jo8l4dfds2vupar7bidbklbe
Found movie: Living (2023)
No matching subtitles were found.
5. https://pastebin.com/RbQZ2sN2

**Comment by MichaelYochpaz** [comment]
I've created a new branch with logs to code sections that are related to this issue, to allow me to further investigate this issue.

Please run `python3 -m pip install -e git+https://github.com/MichaelYochpaz/iSubRip.git@issue-43-logs#egg=isubrip` to install it, and then rerun the command (`isubrip https://tv.apple.com/fi/movie/goldstone/umc.cmc.5up0cp30rbmrod9hc1xjq86o3?playableId=tvs.sbd.9001%3A1501840635`) and send me the output (you should see a bunch of `LOG:` lines prints)

Also, please add it in a formatted-code format as it should be a somewhat-long output, and GitHub might mess it up if you don't.
Paste it in-between triple backticks, like the following example:
````
```
< Logs go here >
```
````

After that, you can uninstall iSubRip and reinstall it normally to return to the base version without the logs.

**Comment by williameyelash** [comment]
https://pastebin.com/VCd1A0hr

**Comment by MichaelYochpaz** [comment]
Thanks for the logs.
There seems to be a small section of code where errors are being suppressed, which is why it was hard to track down up to this point (no error was raised).

I've fixed that on the debug branch (I will also update that on `main` for the next version release), please rerun the pip installation command to update it:
```
python3 -m pip install -e git+https://github.com/MichaelYochpaz/iSubRip.git@issue-43-logs#egg=isubrip`
```

And then rerun the iSubRip command and send me the new logs (hopefully for the last time).

Looking at the relevant code section that was suppressed, I suspect this issue might be path related (this section generates a file name, and saves the subtitles to disk on a temp folder), so you could also try running the command on a different location (for example `C:\Test`), or set a path in the config and see if the behavior changes.

Sorry for all the hassle.

**Comment by williameyelash** [comment]
Here you go: https://pastebin.com/LQy2kHvk

Set a different path in the config.toml that's in Users\Username\.isubrip but nothing changed.

I wonder if anyone else is having this issue.

**Comment by MichaelYochpaz** [comment]
Thanks for the logs.
I was finally able to find out what caused this issue, and successfully reproduce it :)

It's caused by a [bug on Python version 3.11.1](https://github.com/python/cpython/issues/100098), that was later fixed on version 3.11.2 (this is from the [changelog](https://docs.python.org/3/whatsnew/changelog.html#python-3-11-2-final)):
```
gh-100098: Fix tuple subclasses being cast to tuple when used as enum values.
```

I've been using Python 3.11.3 which is why I couldn't reproduce it.  
I've installed 3.11.1 (the version you're using), and the error reproduced.

So basically, if you'll update your Python version (or downgrade it) this should be resolved.
I'm leaving the issue open, please report back if you were able to fix it.

**Comment by williameyelash** [comment]
Works fine now, thanks!
